android/capacitor/src/main/java/com/getcapacitor/plugin/notification/DateMatch.java

```diff
@@ -92,46 +92,47 @@ public long nextTrigger(Date date) {
    * Postpone trigger if first schedule matches the past
    */
   private long postponeTriggerIfNeeded(Calendar current, Calendar next) {
-    int currentYear = current.get(Calendar.YEAR);
-    if (matchesUnit(Calendar.YEAR, current, next)) {
-      next.set(Calendar.YEAR, currentYear + 1);
-      this.unit = Calendar.YEAR;
-    } else if (matchesUnit(Calendar.MONTH, current, next)) {
-      next.set(Calendar.YEAR, currentYear + 1);
-      this.unit = Calendar.MONTH;
-    } else if (matchesUnit(Calendar.DAY_OF_MONTH, current, next)) {
-      next.set(Calendar.MONTH, current.get(Calendar.MONTH) + 1);
-      this.unit = Calendar.DAY_OF_MONTH;
-    } else if (matchesUnit(Calendar.HOUR_OF_DAY, current, next)) {
-      next.set(Calendar.DAY_OF_MONTH, current.get(Calendar.DAY_OF_MONTH) + 1);
-      this.unit = Calendar.DAY_OF_MONTH;
-    } else if (matchesUnit(Calendar.MINUTE, current, next)) {
-      next.set(Calendar.HOUR_OF_DAY, current.get(Calendar.HOUR_OF_DAY) + 1);
-      this.unit = Calendar.MINUTE;
+    if (next.getTimeInMillis() <= current.getTimeInMillis() && unit != -1) {
+      Integer incrementUnit = -1;
+      if (unit == Calendar.YEAR || unit == Calendar.MONTH) {
+        incrementUnit = Calendar.YEAR;
+      } else if (unit == Calendar.DAY_OF_MONTH) {
+        incrementUnit = Calendar.MONTH;
+      } else if (unit == Calendar.HOUR_OF_DAY) {
+        incrementUnit = Calendar.DAY_OF_MONTH;
+      } else if (unit == Calendar.MINUTE) {
+        incrementUnit = Calendar.HOUR_OF_DAY;
+      }
+
+      if (incrementUnit != -1) {
+        next.set(incrementUnit, next.get(incrementUnit) + 1);
+      }
     }
+    
```

Comment:
I don't think it's related to your change, but I've noticed that on Android, when using on, the notification is fired at the same second it was scheduled, while on iOS it's fired on second 0.
Example, I want to fire a notification at 19:10
On iOS it fires at 19:10:00
On Android it fires at 19:10:XY, where XY can be any second based on the second it was scheduled, so can be any value from 00 to 59, there could be almost 1 minute of difference.

I think it makes more sense iOS behavior, if you set it to "minute 10" it should fire at exact 10 minute and 0 seconds, not to an unpredictable second. 

I've accomplished that by adding this line

```suggestion
    next.set(Calendar.SECOND, 0);
```

If you think it makes sense, accept the suggestion.